### ISSUE TITLE
Add S5-GC30K support

### ISSUE THREAD

**Issue by poligor84** [StephanJoubert/home_assistant_solarman#654]
Good day.
Please add support for my grid inverter S5-GC30K support.
This is my inverter in Solis cloud:
![1](https://github.com/user-attachments/assets/d396c3fd-d73a-4baf-8596-d6ce46aff01f)
and not all sensors are displayed in your application:
![2](https://github.com/user-attachments/assets/05014140-ccf9-496d-93ac-dd894967ecdc)
![3](https://github.com/user-attachments/assets/e4020c21-399d-4211-b64c-d0a2bc823a08)
Thank you.

**Comment by davidrapan** [comment]
Hi @poligor84 did you tried load it with every profile there?

**Comment by poligor84** [comment]
> Hi @poligor84 did you tried load it with every profile there?

What do you mean by "profile"?

**Comment by poligor84** [comment]
> Hi @poligor84 did you tried load it with every profile there?

![4](https://github.com/user-attachments/assets/ce3304c1-db91-495d-88d7-44a46e27a075)

**Comment by davidrapan** [comment]
yes, I would suggest you to try it with every one of them...

**Comment by poligor84** [comment]
> Hi @poligor84 did you tried load it with every profile there?

![5](https://github.com/user-attachments/assets/f376f6b1-0f9f-4134-b436-03b3ca8aac51)

**Comment by davidrapan** [comment]
yes all the solis, sofar and zcs are in some cases very similar...

**Comment by poligor84** [comment]
> yes all the solis, sofar and zcs are in some cases very similar...

Tried. The situation in general has not improved!

**Comment by davidrapan** [comment]
Uhh, so you can also try this integration: [davidrapan/ha-solarman](https://github.com/davidrapan/ha-solarman) there is even more of them... otherwise it has to be added as a new one then, you could try to ask your manufacturer to give you modbus rtu documentation for your device...

**Comment by poligor84** [comment]
![6](https://github.com/user-attachments/assets/c5ca097e-a340-4ca5-aaf7-27e1e2582cb0)
![7](https://github.com/user-attachments/assets/3ba92a69-a2cc-4964-bf17-f99021dbd148)

**Comment by poligor84** [comment]
> Uhh, so you can also try this integration: [davidrapan/ha-solarman](https://github.com/davidrapan/ha-solarman) there is even more of them... otherwise it has to be added as a new one then, you could try to ask your manufacturer to give you modbus rtu documentation for your device...

Ok. one second....

**Comment by davidrapan** [comment]
The profile is missing PV3 entities. 😉 I can try to add them in my integration if you help me locate those registers. 😉

**Comment by poligor84** [comment]
> Uhh, so you can also try this integration: [davidrapan/ha-solarman](https://github.com/davidrapan/ha-solarman) there is even more of them... otherwise it has to be added as a new one then, you could try to ask your manufacturer to give you modbus rtu documentation for your device...

![8](https://github.com/user-attachments/assets/382d8b83-ad7a-4693-b1ae-f936cbc0b5a7)
![9](https://github.com/user-attachments/assets/4f9d9456-aa28-4626-942a-336bf426eccd)
![91](https://github.com/user-attachments/assets/b32a6bbf-af92-4716-804f-7b43eb0d2148)
![92](https://github.com/user-attachments/assets/aa8048b3-4edf-444d-9524-a68a7faaa384)
I can't join for some reason?!

**Comment by davidrapan** [comment]
You have to disable the passthrough mode! It's only for ethernet sticks!

**Comment by poligor84** [comment]
пн, 9 вер. 2024 р. о 11:23 David Rapan ***@***.***> пише:

**Comment by davidrapan** [comment]
You have to disable the Solarman protocol passthrough mode! It's only for ethernet sticks!

**Comment by poligor84** [comment]
![93](https://github.com/user-attachments/assets/d42c2a07-ae9b-4bc0-8a34-f5c79d39dbf4)

**Comment by davidrapan** [comment]
Yes.

**Comment by poligor84** [comment]
to disable

пн, 9 вер. 2024 р. о 11:30 David Rapan ***@***.***> пише:

**Comment by poligor84** [comment]
> Yes.

no work

**Comment by davidrapan** [comment]
Click on enable debug logging, try to add device. Wait few seconds and then click on it again. It will download a file. Post the file here please.

**Comment by poligor84** [comment]
пн, 9 вер. 2024 р. о 11:35 David Rapan ***@***.***> пише:

**Comment by poligor84** [comment]
![94](https://github.com/user-attachments/assets/cbdbf3e7-c217-4b6d-8be4-54ace02a72ff)

**Comment by davidrapan** [comment]
?

**Comment by poligor84** [comment]
> ?

Maybe I have a conflict between two versions? and can't connect?

**Comment by davidrapan** [comment]
The one you downloaded as last is "active" so no. I told you what to do so we can find out what's wrong. I need the debug logs please. 😉

**Comment by poligor84** [comment]
> The one you downloaded as last is "active" so no. I told you what to do so we can find out what's wrong. I need the debug logs please. 😉

[home-assistant_solarman_2024-09-09T09-18-59.702Z.log](https://github.com/user-attachments/files/16928371/home-assistant_solarman_2024-09-09T09-18-59.702Z.log)

**Comment by davidrapan** [comment]
Yeah it has problems with some requested registers. You have to try use some different profile.

**Comment by poligor84** [comment]
> Yeah it has problems with some requested registers. You have to try use some different profile.

Thanks, working on the profile 3P-4G!

**Comment by poligor84** [comment]
![95](https://github.com/user-attachments/assets/49c7207e-fbc2-4a6c-8d93-689601812cf1)

**Comment by davidrapan** [comment]
Wait for new version (I already committed the file if you know how to "replace" it manually) where will be added profile with more PV inputs. 😉

**Comment by poligor84** [comment]
Ok, thanks.

пн, 9 вер. 2024 р. о 20:59 David Rapan ***@***.***> пише:

**Comment by davidrapan** [comment]
@poligor84 Beta version [v24.09.09](https://github.com/davidrapan/ha-solarman/releases/tag/v24.09.09) contains profile named: solis_3p-4g+.yaml. You have to allow installing beta versions in HACS under 3 dots menu.

**Comment by poligor84** [comment]
Thanks, I will!
[image: image.png]

It works!
Thank you!!!


вт, 10 вер. 2024 р. о 00:19 David Rapan ***@***.***> пише:

**Comment by jedifr** [comment]
Hello,
I have the same inverter, but my connection key is WIFI.
My concern is that I am not able to insert the serial number of the key because it does not match the expected number type.
My number is type: 5A12338417B0AFF6

Do you have an idea?
THANKS

**Comment by poligor84** [comment]
I don't quite understand what you mean?
You can't connect to HA?

**Comment by jedifr** [comment]
I can't introduice my serial number key with good numers like 5A12338417B0AFF6
The app change automatiquely the serial number...

![Image](https://github.com/user-attachments/assets/ccf50691-acb6-44ad-ac06-d35661c2a2a6)
